packages/babel-parser/src/parser/expression.js

```diff
@@ -1940,6 +1943,11 @@ export default class ExpressionParser extends LValParser {
       if (refTrailingCommaPos && this.match(tt.comma)) {
         refTrailingCommaPos.start = this.state.start;
       }
+    } else if (allowPlaceholder && this.match(tt.question)) {
```

Comment:
Wdyt about this logic, to have better errors?

```js
} else if (this.match(tt.question)) {
  this.expectPlugin("partialApplication");
  if (allowPlaceholder) {
    this.raise(this.state.start, "Unexpected argument placeholder");
  }
  // ...
```

Reply:
I like this! It has a clear message and does not confuse the user. The user will understand that they should not use `ArgumentPlaceholder` there. I'll apply it in my next push.